src/Conversion/KrnlToLLVM/KrnlToLLVMHelper.cpp

```diff
@@ -115,59 +116,8 @@ int64_t getRankFromMemRefType(LLVM::LLVMStructType memRefTy) {
 }
 
 // Convert an MLIR type to the correspoding ONNX type.
-onnx::TensorProto::DataType mlirTypeToOnnxType(Type elemType) {
-  onnx::TensorProto::DataType onnxType = onnx::TensorProto::UNDEFINED;
-
-  TypeSwitch<Type>(elemType)
-      .Case<BFloat16Type>(
-          [&](BFloat16Type) { onnxType = onnx::TensorProto::BFLOAT16; })
-      .Case<ComplexType>([&](ComplexType type) {
-        if (type.getElementType().isa<Float32Type>())
-          onnxType = onnx::TensorProto::COMPLEX64;
-        else if (type.getElementType().isa<Float64Type>())
-          onnxType = onnx::TensorProto::COMPLEX128;
-      })
-      .Case<Float16Type>(
-          [&](Float16Type) { onnxType = onnx::TensorProto::FLOAT16; })
-      .Case<Float32Type>(
-          [&](Float32Type) { onnxType = onnx::TensorProto::FLOAT; })
-      .Case<Float64Type>(
-          [&](Float64Type) { onnxType = onnx::TensorProto::DOUBLE; })
-      .Case<IntegerType>([&](IntegerType type) {
-        switch (type.getWidth()) {
-        case 1:
-          // only a signless type can be a bool.
-          onnxType = (type.isSigned() || type.isUnsigned())
-                         ? onnx::TensorProto::UNDEFINED
-                         : onnx::TensorProto::BOOL;
-          break;
-        case 8:
-          onnxType = type.isUnsigned() ? onnx::TensorProto::UINT8
-                                       : onnx::TensorProto::INT8;
-          break;
-        case 16:
-          onnxType = type.isUnsigned() ? onnx::TensorProto::UINT16
-                                       : onnx::TensorProto::INT16;
-          break;
-        case 32:
-          onnxType = type.isUnsigned() ? onnx::TensorProto::UINT32
-                                       : onnx::TensorProto::INT32;
-          break;
-        case 64:
-          onnxType = type.isUnsigned() ? onnx::TensorProto::UINT64
-                                       : onnx::TensorProto::INT64;
-          break;
-        }
-      })
-      .Case<LLVM::LLVMStructType>(
-          [&](LLVM::LLVMStructType) { onnxType = onnx::TensorProto::STRING; });
-
-  if (onnxType == onnx::TensorProto::UNDEFINED) {
-    elemType.dump();
-    llvm_unreachable("MLIR type cannot be converted to ONNX type");
-  }
-
-  return onnxType;
+int64_t mlirTypeToOnnxType(Type elemType) {
```

Comment:
Can the krnl::mlirTypeToOnnxType be replaced with onnx_mlir::mlirTypeToOnnxType?

Reply:
I got the impression that the src/Dialect/Krnl source code tries to avoid dependencies on ONNX and therefore I put this dependency on ONNX inside the implementation file KrnlToLLVMHelper.cpp, hiding it from the use in KrnlEntryPoint.cpp which just includes the header file KrnlToLLVMHelper.hpp

so this PR creates a half-hidden dependency of Krnl on ONNX which is not ideal, I'm open to suggestions to do it differently

FWIW, AFAICT the RuntimeAPI SET/GET_DATA_TYPE calls in `krnl::fillOMTensorWithMemRef()` and `KrnlEntryPointOpLowering::emitVerificationCodeForInputTensors()` just want any one-to-one mapping of MLIR data types to i64 and doesn't care that the i64 numbers match the onnx::TensorProto enum